specify/specify7
# Allow consistent behavior for Reports/Labels and improve Report preferences

Fixes #3570 

The underlying cause of this issue is a preference which cleared all query filters when running a report or label from a Form. This behavior was first implemented in https://github.com/specify/specify7/commit/199c11697ed0fb8cafee436888a4f9f259537cbf and later made into a preference in https://github.com/specify/specify7/commit/e1faeb2d0f200ec3084e5ae16bfa1c0b7c8efaf4. 

This preference has a **_significant_** impact to the behavior of reports. For example, if the query (with a CollectionObject base table) for the report had the fields `determinations -> isCurrent -> Equals -> True` and `preparations -> preptype -> name -> Equals -> SomePrepType`, these would be passed to the report runner as `determinations -> isCurrent -> Equals -> Any` and `preparations -> preptype -> name -> Equals -> Any` (respectively). 

@specify/ux-testing, what is the benefit for this behavior? Why was it implemented in such a way in the first place? I severely doubt this is wanted by users and is definitely not expected.

Thankfully for those currently struggling with #3570, the remedy of disabling the preference should suffice for most cases without having to update.

In this PR, I have set the default value of the preference to False, provided a better description for the preference, and defined a new preference which would display the query filters and not auto-run the Report or Label from the Form. 

### Testing Instructions 
- Identify a label/report which has an associated Query with any filters (such as the [Small Labels](https://fwri6123-issue-3570.test.specifysystems.org/specify/resources/app-resource/112/) which has a Query field which only returns COs which have "Small"  in the remarks field-localized as Label size- on preparation)
- With the `Reset query filters` preference disabled, locate a record which would normally not return any results with the associated Query (If using the Small Labels report from the previous instruction, an example of such a record is [CO with ID 136663](https://fwri6123-issue-3570.test.specifysystems.org/specify/view/collectionobject/136663/)) 
- Ensure that no results are returned. 
- Modify the record so it would normally be returned by the Query (in the previous example, this would be adding a preparation which has a Small Label size, or modifying already existing preparations so they have a Small Label size)
- Generate the report/label with the modified record and confirm a correct/expected output. 

- - - 

- Ensure the new preference `Show query filters when running a Report from a Form` behaves correctly (once the report/label is selected, the Query Parameters dialog should be displayed and the query should be able to be temporarily modified before being passed to the report runner. Note the query should have an additional field at the bottom of the Query with an ID field; this is expected).

- An inconsistency in reports/labels was also demonstrated in https://github.com/specify/specify7/issues/3141#issuecomment-1569495107. Can this be tested as well?
